src/main/java/org/opensearch/security/user/User.java

```diff
@@ -290,10 +290,17 @@ public final Set<String> getSecurityRoles() {
     /**
      * Check the custom attributes associated with this user
      *
-     * @return true if it has a service account attributes. otherwise false
+     * @return true if it has a service account attributes, otherwise false
      */
     public boolean isServiceAccount() {
         Map<String, String> userAttributesMap = this.getCustomAttributesMap();
         return userAttributesMap != null && "true".equals(userAttributesMap.get("attr.internal.service"));
     }
+
+    /**
+     * @return true if this instance is of the type PluginUser
+     */
+    public boolean isPluginUser() {
```

Comment:
This goes against the pattern of polymorphism, the base class shouldn't be updated to know if it is a subclass.  This forces callers of User to determine what subclasses are supported.

Reply:
Changed to using `instanceof`. I'll look into other ways to do this as well, but I'm reticent to user the same user attributes approach that we used with service accounts.

Reply:
Can you help me understand why not?  To take this to an extreme, if a user object representing plugin could have bad interactions with existing scenarios, why re-use the User object at all?